plugin/inlay_hint.py

```diff
@@ -162,7 +166,10 @@ def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uui
                 }
             })
             value += f'<a href="{inlay_hint_click_command}">'
-        value += html.escape(label_part['value'])
+        raw_label = label_part['value']
+        truncated_label = raw_label[:truncate_limit] + '...' if len(raw_label) >= truncate_limit else raw_label
```

Comment:
`if len(raw_label) == truncate_limit` then we shouldn't be adding elipsis

Reply:
probably just use textwrap.shorten(...)

Reply:
It looks like that one seems to collapse spaces but maybe that's OK...

> First the whitespace in text is collapsed (all whitespace is replaced by single spaces)

https://docs.python.org/3/library/textwrap.html#textwrap.shorten

Reply:
> It looks like that one collapses spaces but maybe that's OK...
> 
> > First the whitespace in text is collapsed (all whitespace is replaced by single spaces)
> 
> https://docs.python.org/3/library/textwrap.html#textwrap.shorten

I made a quick prototype with `textwrap.shorten`, but it drops whitespace, and it affects the readability of the inlay hint:


<h6 align="center">With `textwrap.shorten`</h6>

![textwrap-shorten-usage-example](https://github.com/user-attachments/assets/7c1c9746-43e1-4868-8964-51260d45e060)


<h6 align="center">With manual implementation</h6>

![manual-truncation-usage-example](https://github.com/user-attachments/assets/3e40a551-78fb-4be7-a7c6-702927c143eb)

I think I will stick with manual truncation because it is easier to manage in the for loop. I think `textwrap.shorten` might not work properly if we're using "parts" of the label.